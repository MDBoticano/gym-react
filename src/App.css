#root {
  height: 100vh;
  background-color: hsl(0, 0%, 97%);
  margin: 0;
  padding: 0;
}

h1 {
  margin: 0;
}