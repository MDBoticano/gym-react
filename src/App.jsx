import React from 'react';

import Navigation from './components/Navigation';
import Exercises from './components/Exercises';
import Card from './components/Card';
import Tag from './components/Tag';

const App = () => {
  return (
    <div className="App">
      <Exercises>

      <Card 
        title={"Push Ups"}
        description={"This is a description of what the exercise entails or does not or what is important and necessarr…"}
      >
        <Tag text={"Hello"} />
      </Card>

      <Card 
        title={"Sit Ups"}
        description={"This is a description of what the exercise entails or does not or what is important and necessarr…"}
      >
        <Tag text={"Abs"} />
      </Card>
      </Exercises>
      <Navigation />
    </div>
  );
};

export default App;
