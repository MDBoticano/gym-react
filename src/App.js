import React, { useState, useEffect } from 'react'
import './App.css'
import Filter from './Filter'
import Exercises from './Exercises'
import AddExercise from './AddExercise'
import exerciseService from './services/exerciseService'

const App = () => {
  const defaultFilterTypes = {
    "name": true,
    "category": true
  }
  const [filterTerm, setFilterTerm] = useState('')
  const [filterTypes, setFilterTypes] = useState(defaultFilterTypes)
  const [exerciseList, setExerciseList] = useState([])

  /* From form */
  const [formName, setFormName] = useState('')
  const [formCategories, setFormCategories] = useState('')

  /* GET: Retreive list of exercises from server */
  useEffect(() => {
    exerciseService
      .getExercises()
      .then((retreievedExercises) => {
        setExerciseList(retreievedExercises)
      })
  }, [])

  const filterList = (event) => {
    setFilterTerm(event.target.value)
  }

  const filtersSelector = (type) => {
    setFilterTypes({ ...filterTypes, [type]: !filterTypes[type] })
  }

  const chipSetsFilter = (chipName) => {
    /* Set sort by categories to be true */
    setFilterTypes({...filterTypes, "category": true})
    setFilterTerm(chipName)
  }

  const makeId = () => {
    /* +2 since IDs are 1-indexed */
    return exerciseList.length + 2
  }

  const submitExercise = (event) => {
    event.preventDefault()
    console.log('Name:', formName)
    console.log('Categories:', formCategories)

    const splitCategories = formCategories.split(' ')
    console.log(splitCategories)

    const exerciseEntry = {
      name: formName,
      id: makeId(),
      category: splitCategories,
      date: new Date().toISOString()
    }

    exerciseService
      .createExercise(exerciseEntry)
      .then(returnedEntry => {
        setExerciseList(exerciseList.concat(returnedEntry))
      })
      .then(console.log('Successfully added entry'))
      .catch(error => console.log(error.response.data))

    /* clear */
    setFormName('')
    setFormCategories('')
  }

  const nameHandler = (event) => {
    setFormName(event.target.value)
  }

  const categoriesHandler = (event) => {
    setFormCategories(event.target.value)
  }

  return (
    <div id="App">
      <Filter 
        filterTypes={filterTypes}
        filtersSelector={filtersSelector} 
        filterTerm={filterTerm}  
        filterHandler={filterList} 
      />
      <Exercises
        filterBy={filterTypes} 
        filterTerm={filterTerm}
        exerciseList={exerciseList}
        chipSetsFilter={chipSetsFilter}
      />
      <AddExercise
        submitExercise={submitExercise} 
        nameHandler={nameHandler} formName={formName}
        categoriesHandler={categoriesHandler} formCategories={formCategories}
      />
    </div>
  )
}

export default App