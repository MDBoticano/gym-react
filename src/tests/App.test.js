import React from 'react';
import App from '../App';
import { create } from 'react-test-renderer';

test('App is app', () => {
  const component = create(
    <App />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});