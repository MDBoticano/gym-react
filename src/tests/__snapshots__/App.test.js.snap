// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is app 1`] = `
<div
  className="App"
>
  <div
    className="sc-gPEVay eHieyz"
  >
    <header
      className="sc-iRbamj dZnHBM"
    >
      <p
        className="page-title"
      >
        Exercises
      </p>
      <div
        className="sc-bdVaJa kMyqSV ExercisesSearch"
      >
        <input
          className="sc-bwzfXH eAKyFD"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </header>
    <div
      className="sc-jlyJG fflAYx ExercisesList"
    >
      <div
        className="sc-gipzik ilydas"
      >
        Retrieving exercises...
      </div>
    </div>
    <button
      className="sc-htpNat uapYE"
      onClick={[Function]}
    >
      Create an Exercise
    </button>
  </div>
</div>
`;
